lib/sycamore/sycamore/plan_nodes.py

```diff
@@ -199,3 +200,38 @@ def __init__(self, child: Node, **resource_args):
 
     def __str__(self):
         return "write"
+
+
+def print_plan(node: Node, stream=sys.stdout) -> None:
+    """Utility function for printing plans.
+
+    Prints the node and all its children. Indentation is used
+    to indicate the parent-child relation, e.g.
+
+    RootNode
+      InternalNode1
+        LeafNode1
+      InternalNode2
+        LeafNode2
+    """
+
+    class PrintTraverse(NodeTraverse):
+        def __init__(self):
+            super().__init__()
+            self.indent = 0
+            self.stream = stream
+            self.before_fn = self._print_node
```

Comment:
this works, but you can remove the self.before_fn/after_fn setting and write the functions as:
```
def before(self, n: Node) -> Node:
   ... code for print_node ...

def after(self, n: Node) -> Node:
  ... code in after_node ...
```